version: "3.8"

volumes:
  vol_prod:
    driver: local
    driver_opts:
      type: nfs
      o: addr=10.0.2.10,rw,noatime,rsize=8192,wsize=8192,tcp,timeo=14,nfsvers=4
      device: ":/volume1/nfs_docker_gitea"

networks:
  clra_macvlan:
    external: true

services:
  gitea:
    image: gitea/gitea:latest
    container_name: gitea
    networks:
      clra_macvlan:
        ipv4_address: 10.0.2.22
    #ports:
    #  - "3000:3000"
    #  - "22:22"
    volumes:
      - vol_prod:/data
    environment:
      - DB_TYPE=mysql
      - DB_HOST=10.0.2.10:3306
      - DB_NAME= "[Gitea Database Name]"
      - DB_USER= "[Gitea User Name]"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.gitea.rule=Host(`gitea.homelab.com`)"
      - "traefik.http.routers.gitea.entrypoints=websecure"
      - "traefik.http.routers.gitea.tls=true"
      #- "traefik.http.routers.gitea.tls.certresolver=production" #Using wildcard from Traefik3
      - "traefik.http.routers.gitea.service=gitea"
      - "traefik.http.services.gitea.loadbalancer.server.port=3000"
    restart: always